source/GameWindow.cpp

```diff
@@ -402,6 +418,25 @@ void GameWindow::ToggleFullscreen()
 
 
 
+int GameWindow::CurrentDisplayID()
+{
+	int posX = 0, posY = 0, ID = 0;
+	SDL_GetWindowPosition(mainWindow, &posX, &posY);
+	for(int i = 0; i < SDL_GetNumVideoDisplays(); i++)
+	{
+		SDL_Rect currentDisplay;
+		SDL_GetDisplayBounds(i, &currentDisplay);
+		if(posX >= currentDisplay.x && posX < currentDisplay.w && posY >= currentDisplay.y && posY < currentDisplay.h)
+		{
+			ID = i;
+			break;
+		}
+	}
+	return ID;
```

Comment:
Does it make sense to react to the "window moved" event (which sends the new XY as event data) to set the new display event?
We'd cache the bounds of each display on startup so we can simply update.
We could then also re-use the same position the next time the game launches, rather than always recentering.

Reply:
Well there's a failsafe, which checks if you actually have the display on which it has to launch the window, and if it's removed it would default back to display 0. That wouldn't work if you just launch on XY coordinates where it was closed...

Reply:
Triggering the failsafe should reset the coordinates

Reply:
Yes but how do you trigger this if you only have coordinates?
Screen::GetDisplayID() >= SDL_GetNumVideoDisplays()

Reply:
That's why I mention caching the bounds of the available displays. We will still need the display ID in order to get the correct display mode and high-dpi information. The event handler would update the stored coordinates and check if the display changed. The display change handler would update the display ID, high DPI state, and window bounds (and any other display state I'm not thinking of).

Reply:
Yeah, I think I'll leave that to you guys. It's getting late, and tomorrow I have other things to do... I kinda just fixed what annoyed me, and don't really want to go down that rabbit hole. Feel free to do whatever you want with the code. :)